test-utils/src/rpc/client.rs

```diff
@@ -0,0 +1,275 @@
+            .unwrap();
+
+        // Make sure this is a trusted, `pow_hash` only works there.
+        assert!(
+        	!result.block_header.pow_hash.is_empty(),
+        	"untrusted node detected, `pow_hash` will not show on these nodes - use a trusted node!"
+        );
+
+        let (block_hash, block) = spawn_blocking(|| {
+            let block = Block::read(&mut hex::decode(result.blob).unwrap().as_slice()).unwrap();
+            (block.hash(), block)
+        })
+        .await
+        .unwrap();
+
+        let txs = self
+            .get_transaction_verification_data(&block.txs)
+            .await
+            .map(Arc::new)
+            .collect();
+
+        let block_header = result.block_header;
+        let block_hash_2 = <[u8; 32]>::try_from(hex::decode(&block_header.hash).unwrap()).unwrap();
+        let pow_hash = <[u8; 32]>::try_from(hex::decode(&block_header.pow_hash).unwrap()).unwrap();
+
+        // Assert the block hash matches.
+        assert_eq!(block_hash, block_hash_2);
+
+        VerifiedBlockInformation {
+            block,
+            txs,
+            block_hash,
+            pow_hash,
+            height: block_header.height,
+            generated_coins: block_header.reward,
+            weight: block_header.block_weight,
+            long_term_weight: block_header.long_term_weight,
+            cumulative_difficulty: block_header.cumulative_difficulty,
+        }
+    }
+
+    /// Request data and map the response to a [`TransactionVerificationData`].
+    ///
+    /// # Panics
+    /// This function will panic at any error point, e.g.,
+    /// if the node cannot be connected to, if deserialization fails, etc.
+    pub async fn get_transaction_verification_data<'a>(
+        &self,
+        tx_hashes: &'a [[u8; 32]],
+    ) -> impl Iterator<Item = TransactionVerificationData> + 'a {
+        self.rpc
+            .get_transactions(tx_hashes)
+            .await
+            .unwrap()
+            .into_iter()
+            .enumerate()
+            .map(|(i, tx)| {
+                let tx_hash = tx.hash();
+                assert_eq!(tx_hash, tx_hashes[i]);
+                TransactionVerificationData {
+                    tx_blob: tx.serialize(),
+                    tx_weight: tx.weight(),
+                    tx_hash,
+                    fee: tx.rct_signatures.base.fee,
+                    tx,
+                }
+            })
+    }
+}
+
+//---------------------------------------------------------------------------------------------------- TESTS
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use hex_literal::hex;
+
+    /// Assert the default address is localhost.
+    #[tokio::test]
+    async fn localhost() {
+        assert_eq!(HttpRpcClient::new(None).await.address(), LOCALHOST_RPC_URL);
+    }
+
+    /// Assert blocks are correctly received/calculated.
+    #[ignore] // FIXME: doesn't work in CI, we need a real unrestricted node
+    #[tokio::test]
+    async fn get() {
```

Comment:
TODO: It would be nice if our CI had access to `monerod` RPC calls.

We could make our current CI `monerod` forward RPC calls from other nodes but unfortunately [I believe getting `pow_hash` in the response requires an unrestricted node](https://github.com/monero-project/monero/blob/c8214782fb2a769c57382a999eaf099691c836e7/src/rpc/core_rpc_server.cpp#L2615) so maybe we could get our own 24/7 node for testing purposes? `username:password` would be passed via CI env variables I guess.

Reply:
We could but this is test code, for the test code so not high importance IMHO, probably can leave this test to be run every so often manually.